### PR TITLE
Attempt at a cleaner formalization of the PRId algorthm

### PR DESCRIPTION
**Pull request by wilwade**
Trying a different way to specify the PRId Algorithm

### CODE REVIEW

**Review by shannonwells — APPROVED** [review]
WAY better!

File: pages/DSNP/Graph.md [hunk 1]
```diff
@@ -108,20 +108,38 @@ In the following section, the Alice to Bob identifier for context C is called PR
...
+    * Libsodium: [`crypto_kdf_derive_from_key`](https://libsodium.gitbook.io/doc/key_derivation)
+    * <code>Ctx<sub>KDF</sub> = "PRIdCtx0"</code>
+    * <code>CtxSharedSecret<sub>A→B</sub> = <abbr title="Key Derivation Function">KDF</abbr>(Id<sub>B</sub> Ctx<sub>KDF</sub>, RootSharedSecret<sub>AB</sub>)</code>
 1. Alice uses Bob's DSNP User Id to form a 8-byte message.
-1. Alice encrypts the message using the PRId key <code>k<sub>Bob</sub></code> and a nonce of her own User Id (padded with zero bytes), as in the libsodium function `crypto_secretbox_easy`.
-1. Alice removes the message authentication code (MAC), leaving the 8-byte value <code>PRId<sub>A→B,C</sub></code>.
-  (Because Alice will publish the PRId, it can be authenticated without a MAC.)
+    * <code>MSG<    sub>A→B</sub> = Id<sub>B</sub></code>
+1. Alice encrypts the message using the PRId key <code>CtxSharedSecret<sub>A→B</sub></code> and a nonce of her own User Id (padded with zero bytes).
```
Comment (aramikm) [inline pages/DSNP/Graph.md:131]: what is the byte size of this padded nonce value? it would be nice to have an example of padding

**Review by aramikm — APPROVED** [review]
So much better! Added 2 nits.

File: pages/DSNP/Graph.md (hunk 1 above)
Comment (wilwade) [reply pages/DSNP/Graph.md:131]: It is 24 bytes for XSalsa20 adding that in...

Comment (wilwade) [reply pages/DSNP/Graph.md]: Updated. No example, but a specific endianness as well so much better defined.

File: pages/DSNP/Graph.md [hunk 2]
```diff
@@ -108,20 +108,38 @@ In the following section, the Alice to Bob identifier for context C is called PR
...
+    * Libsodium: [`crypto_box_beforenm`](https://libsodium.gitbook.io/doc/public-key_cryptography/authenticated_encryption#precalculation-interface)
+    * <code>RootSharedSecret<sub>AB</sub> = <abbr title="Elliptic-curve Diffie-Hellman">ECDH</abbr>(B<sub>public</sub>, A<sub>private</sub>)</code>
+1. Alice derives a context-specific subkey <code>CtxSharedSecret<sub>Bob</sub></code> from the shared secret `RootSharedSecret` as the master key, Bob's DSNP User Id as the 64-bit key identifier, and the ASCII encoding of the [PRId Context](#contexts) string (`"PRIdCtx0"` for connections).
+    * Libsodium: [`crypto_kdf_derive_from_key`](https://libsodium.gitbook.io/doc/key_derivation)
+    * <code>Ctx<sub>KDF</sub> = "PRIdCtx0"</code>
+    * <code>CtxSharedSecret<sub>A→B</sub> = <abbr title="Key Derivation Function">KDF</abbr>(Id<sub>B</sub> Ctx<sub>KDF</sub>, RootSharedSecret<sub>AB</sub>)</code>
+1. Alice uses Bob's DSNP User Id to form a 8-byte little-endian message.
+    * <code>MSG<sub>A→B</sub> = Id<sub>B</sub></code>
```
Comment (wesbiggs) [inline pages/DSNP/Graph.md:130]: Would `Msg` be better as parallel to `Ctx`?

File: pages/DSNP/Graph.md [hunk 3]
```diff
@@ -108,20 +108,38 @@ In the following section, the Alice to Bob identifier for context C is called PR
...
-1. Alice encrypts the message using the PRId key <code>k<sub>Bob</sub></code> and a nonce of her own User Id (padded with zero bytes), as in the libsodium function `crypto_secretbox_easy`.
-1. Alice removes the message authentication code (MAC), leaving the 8-byte value <code>PRId<sub>A→B,C</sub></code>.
-  (Because Alice will publish the PRId, it can be authenticated without a MAC.)
+Definitions:
+* <code>Id<sub>A</sub> = _DSNP User Id of A (little-endian)_</code>
+* <code>Id<sub>B</sub> = _DSNP User Id of B (little-endian)_</code>
+
+Algorithm:
+
+1. Both Alice and Bob generate an asymmetric key pair for use with x25519 <abbr title="Elliptic Curve Integrated Encryption Scheme">ECIES</abbr>.
```
[the remark this answers is not in the document]
Comment (wilwade) [reply pages/DSNP/Graph.md:117]: It was hard for me as well, but checking the libsodium docs it appears they mostly use X. Will swap out.

**Review by wesbiggs — APPROVED** [review]

**Pull request merged**